grouper/models/public_key.py

```diff
@@ -18,3 +75,35 @@ class PublicKey(Model):
     public_key = Column(Text, nullable=False)
     fingerprint = Column(String(length=64), nullable=False, unique=True)
     created_on = Column(DateTime, default=datetime.utcnow, nullable=False)
+
+    def tags(self):
+        # type: () -> List[PublicKeyTag]
+        """Returns the list of tags that are assigned to this public key
+
+        Returns:
+            a list that contains all of the PublicKeyTags that are assigned to this public key
+        """
+        mappings = self.session.query(PublicKeyTagMap).filter_by(key_id=self.id).all()
+        return [mapping.tag for mapping in mappings]
+
+    def my_permissions(self):
+        # type: () -> List[Permission]
+        """Returns the permissions that this public key has. Namely, this the set of permissions
+        that the public key's owner has, intersected with the permissions allowed by this key's
+        tags
+
+        Returns:
+            a list of all permissions this public key has
+        """
+        my_perms = self.user.my_permissions()
+        for tag in self.tags():
+            my_perms = _permission_intersection(my_perms, tag.my_permissions())
+
+        return list(my_perms)
+
+    def pretty_my_permissions(self):
```

Comment:
The issue is that we want to apply the formatting to a list of things (the key's permissions), and then provide the resultant formatted list back to another function. I.e. ["{} ({})".format(perm.name, perm.argument) for perm in key.permissions()].
Unfortunately, Jinja doesn't allow list comprehensions, and I was unable to find some way to unfurl the list comprehension into a traditional for loop in the template that would allow for appending to a list. One possible solution might be to try just appending to the HTML entity that we ultimately set to as the list, but I don't know that that'd be a readability/usability improvement.